server/routerlicious/packages/local-server/src/localWebSocketServer.ts

```diff
@@ -46,7 +46,7 @@ export class LocalWebSocket implements IWebSocket {
 		this.events.emit(event, ...args);
 	}
 
-	public emitToRoom(roomId: string, event: string, ...args: any[]) {
+	public async emitToRoom(roomId: string, event: string, ...args: any[]) {
```

Comment:
it looks like the base interface IWebSocket doesn't define a return type for emitToRoom, which defaults to any. so making it a promise also seem incorrect. i think the real fix is to fix the interface, whatever the right return type is, and then ensure all usages are correct as well.

Reply:
on a brief review, i think void is the correct return type, and the assumption that it should be a promise is incorrect

Reply:
To reconcile the usage of emitToRoom.catch() ([here](https://github.com/microsoft/FluidFramework/blob/054480f1868dfb91c3b9270d11aaae9f5533068d/server/routerlicious/packages/lambdas/src/nexus/disconnect.ts#L110C1-L112C23) and [here](https://github.com/microsoft/FluidFramework/blob/054480f1868dfb91c3b9270d11aaae9f5533068d/server/routerlicious/packages/lambdas/src/nexus/connect.ts#L430C1-L432C31)), would it be better to change these to try/catch instead?

Reply:
if it turns out that IWebSocket.emitToRoom can have a void return, specifically that there is not an implementation that needs it to be a promise, then yes, they should move to try catch. 

Right now the usage is inconsistent, some places, like those you call out assume it's a promise, whereas other places assume it's not. Adding consistency is the most important things here, which means explicitly typing the return value on the interface. 

It seems the right return type is either void or promise<void>. I would prefer void unless there is an implementation of the interface that needs it to be a promise, which i didn't see when i briefly reviewed. Once the interface is well defined, you will then need to make the usages consistent, but this should be easy as there should be compile or lint errors where it is wrong